product.py

```diff
@@ -63,6 +63,20 @@ def isUpgradeable(self):
         self.mount_state()
         result = True
         try:
+            # Don't propose to upgrade a BIOS installation with a UEFI installer and conversely
+            tool = PartitionTool(self.primary_disk)
+            existing_boot_mode = constants.TARGET_BOOT_MODE_LEGACY
```

Comment:
moving this to an `else` arm would be more straightforward

Reply:
Done